airflow/api_connexion/endpoints/dataset_endpoint.py

```diff
@@ -120,3 +130,40 @@ def get_dataset_events(
     return dataset_event_collection_schema.dump(
         DatasetEventCollection(dataset_events=events, total_entries=total_entries)
     )
+
+
+@security.requires_access([(permissions.ACTION_CAN_CREATE, permissions.RESOURCE_DATASET)])
+@provide_session
+def post_dataset_event(session: Session = NEW_SESSION) -> APIResponse:
+    """Create an external dataset event. This endpoint is useful if you want to update a dataset and
+    trigger downstream DAG runs from external services.
+    """
+    try:
+        json_body = dataset_change_schema.load(get_json_request_dict())
+    except ValidationError as err:
+        raise BadRequest(detail=str(err))
+    uri = json_body["dataset_uri"]
+    external_source = request.remote_addr
+    user_id = getattr(current_user, "id", None)
+    timestamp = json_body.get("timestamp", datetime.now())
+    timestamp = timestamp.astimezone(timezone.utc)
+    extra = json_body.get("extra", {})
+    dataset_event = dataset_manager.register_external_dataset_change(
+        dataset=Dataset(uri),
+        external_source=external_source,
+        user_id=user_id,
+        timestamp=timestamp,
+        extra=extra,
+        session=session,
+    )
+
+    if dataset_event:
```

Comment:
i wish it threw a helpful exception if there is no dataset instead of using None to signal that.  though i don't think there's anything we can do about that now, can we @uranusjr ?

Reply:
oh wait... this is a new method.  so we could.  but wait do we even need a new method?

Reply:
@michaelmicheal why do we need a new method for this?  could we not add params to `register_external_dataset_change`?

Reply:
since the params are kwargs-only, i reckon we could make task_instance optional.  

and, thankfully, since it accepts `**kwargs`, adding more params at call site in airflow won't break anything for "old" custom dataset managers